extensions/email/src/main/java/org/mule/extension/email/api/retriever/RetrieverConnection.java

```diff
@@ -0,0 +1,196 @@
+     */
+    public RetrieverConnection(EmailProtocol protocol,
+                               String username,
+                               String password,
+                               String host,
+                               String port,
+                               long connectionTimeout,
+                               long readTimeout,
+                               long writeTimeout,
+                               Map<String, String> properties,
+                               TlsContextFactory tlsContextFactory) throws EmailConnectionException
+    {
+        super(protocol, username, password, host, port, connectionTimeout, readTimeout, writeTimeout, properties, tlsContextFactory);
+        try
+        {
+            this.store = session.getStore(protocol.getName());
+            this.store.connect(username, password);
+        }
+        catch (MessagingException e)
+        {
+            throw new EmailConnectionException(format("Error while acquiring connection with the %s store", protocol), e);
+        }
+    }
+
+    /**
+     * Creates a new instance of the of the {@link RetrieverConnection}.
+     *
+     * @param protocol          the desired protocol to use. One of imap or pop3 (and its secure versions)
+     * @param username          the username to establish the connection.
+     * @param password          the password corresponding to the {@code username}.
+     * @param host              the host name of the mail server
+     * @param port              the port number of the mail server.
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout       the socket read timeout
+     * @param writeTimeout      the socket write timeout
+     * @param properties        additional custom properties.
+     */
+    public RetrieverConnection(EmailProtocol protocol,
+                               String username,
+                               String password,
+                               String host,
+                               String port,
+                               long connectionTimeout,
+                               long readTimeout,
+                               long writeTimeout,
+                               Map<String, String> properties) throws EmailConnectionException
+    {
+        this(protocol, username, password, host, port, connectionTimeout, readTimeout, writeTimeout, properties, null);
+    }
+
+
+    /**
+     * Opens and return the email {@link Folder} of name {@code mailBoxFolder}.
+     * The folder can contain Messages, other Folders or both.
+     *
+     * @param mailBoxFolder the name of the folder to be opened.
+     * @param openMode   open the folder in READ_ONLY or READ_WRITE mode
+     * @return the opened {@link Folder}
+     */
+    public Folder getFolder(String mailBoxFolder, int openMode)
+    {
+        try
+        {
+            folder = store.getFolder(mailBoxFolder);
```

Comment:
what happens if I call getFolder() twice with different arguments. Shouldn't I close the current folder before swtiching to the next one?